handsontable/src/plugins/pagination/ui.js

```diff
@@ -0,0 +1,446 @@
+   */
+  install() {
+    if (this.#refs?.container) {
+      return;
+    }
+
+    const elements = html`${TEMPLATE}`;
+    const {
+      container,
+      first,
+      prev,
+      next,
+      last,
+      pageSizeSelect,
+    } = elements.refs;
+
+    this.#refs = elements.refs;
+
+    container.setAttribute('dir', this.#isRtl ? 'rtl' : 'ltr');
+
+    const isDisabled = event => event.currentTarget.disabled;
+
+    first.addEventListener('click', (event) => {
+      if (!isDisabled(event)) {
+        this.runLocalHooks('firstPageClick');
+      }
+    });
+    prev.addEventListener('click', (event) => {
+      if (!isDisabled(event)) {
+        this.runLocalHooks('prevPageClick');
+      }
+    });
+    next.addEventListener('click', (event) => {
+      if (!isDisabled(event)) {
+        this.runLocalHooks('nextPageClick');
+      }
+    });
+    last.addEventListener('click', (event) => {
+      if (!isDisabled(event)) {
+        this.runLocalHooks('lastPageClick');
+      }
+    });
+    pageSizeSelect.addEventListener('change', () => {
+      const value = pageSizeSelect.value === 'auto' ? 'auto' : Number.parseInt(pageSizeSelect.value, 10);
+
+      this.runLocalHooks('pageSizeChange', value);
+    });
+
+    this.setCounterSectionVisibility(false);
+    this.setNavigationSectionVisibility(false);
+    this.setPageSizeSectionVisibility(false);
+
+    if (this.#uiContainer) {
+      this.#uiContainer.appendChild(elements.fragment);
+
+      addClass(container, [this.#themeName, 'handsontable']);
+    } else {
+      this.#rootElement.after(elements.fragment);
+    }
+  }
+
+  /**
+   * Updates the width of the pagination container.
+   *
+   * @param {number} width The new width of the pagination container.
+   * @returns {PaginationUI} The instance of the PaginationUI for method chaining.
+   */
+  updateWidth(width) {
+    this.#refs.container.style.width = `${width}px`;
+
+    return this;
+  }
+
+  /**
+   * Updates the theme of the pagination container.
+   *
+   * @param {string | false | undefined} themeName The name of the theme to use.
+   * @returns {PaginationUI} The instance of the PaginationUI for method chaining.
+   */
+  updateTheme(themeName) {
+    this.#themeName = themeName;
+
+    if (this.#uiContainer) {
+      const { container } = this.#refs;
+
+      removeClass(container, /ht-theme-.*/g);
+
+      if (this.#themeName) {
+        addClass(container, this.#themeName);
+      }
+    }
+
+    return this;
+  }
+
+  /**
+   * Gets the height of the pagination container element.
+   *
+   * @returns {number}
+   */
+  getHeight() {
+    return this.#refs.container.offsetHeight;
+  }
+
+  /**
+   * Refreshes the border state of the pagination container based on the external condition.
+   *
+   * @returns {PaginationUI} The instance of the PaginationUI for method chaining.
+   */
+  refreshBorderState() {
+    const { container } = this.#refs;
+
+    if (this.#uiContainer || this.#shouldHaveBorder()) {
```

Comment:
Should this be fired as often as it does? Currently, in some situations, it fires after every scroll event, which is pretty often.

Reply:
The border for the pagination UI needs to be adjusted for a specific vertical scroll position. That's why the logic that executes on each scroll is here.